verilog/analysis/checkers/proper_parameter_declaration_rule.cc

```diff
@@ -40,25 +42,66 @@ using verible::matcher::Matcher;
 // Register ProperParameterDeclarationRule
 VERILOG_REGISTER_LINT_RULE(ProperParameterDeclarationRule);
 
-static constexpr absl::string_view kParameterMessage =
-    "\'parameter\' declarations should only be within packages or in the "
-    "formal parameter list of modules/classes.";
-static constexpr absl::string_view kLocalParamMessage =
-    "\'localparam\' declarations should only be within modules\' or classes\' "
+static constexpr absl::string_view kParameterNotInPackageMessage =
+    "\'parameter\' declarations should only be in the formal parameter list of "
+    "modules/classes.";
+static constexpr absl::string_view kParameterAllowPackageMessage =
+    "\'parameter\' declarations should only be in the formal parameter list of "
+    "modules and classes or in package definition bodies.";
+static constexpr absl::string_view kLocalParamNotInPackageMessage =
+    "\'localparam\' declarations should only be within modules or class "
     "definition bodies.";
+static constexpr absl::string_view kLocalParamAllowPackageMessage =
+    "\'localparam\' declarations should only be within modules, packages or "
+    "class definition bodies.";
+
+static absl::string_view kParameterMessage = kParameterNotInPackageMessage;
+static absl::string_view kLocalParamMessage = kLocalParamAllowPackageMessage;
 
 const LintRuleDescriptor &ProperParameterDeclarationRule::GetDescriptor() {
   static const LintRuleDescriptor d{
       .name = "proper-parameter-declaration",
       .topic = "constants",
       .desc =
           "Checks that every `parameter` declaration is inside a "
-          "package or in the formal parameter list of modules/classes and "
-          "every `localparam` declaration is inside a module or class.",
-  };
+          "formal parameter list of modules/classes and "
+          "every `localparam` declaration is inside a module, class or "
+          "package.",
+      .param = {
+          {
+              .name = "package_allow_parameter",
+              .default_value = "false",
```

Comment:
I'd probably not change the default in this PR.

If the default is to be changed, it should be done in a separate (1-liner) change referencing the LRM recommendations. That way new behavior can be narrowed to a small change.